diff --git a/configure b/configure
# Patch for ffmpeg release/7.1/7.1.1
--- a/configure
+++ b/configure
@@ -6676,7 +6676,7 @@ written by
 check_lib camera2ndk "stdbool.h stdint.h camera/NdkCameraManager.h" ACameraManager_create -lcamera2ndk

 enabled appkit       && check_apple_framework AppKit
-enabled audiotoolbox && check_apple_framework AudioToolbox
+enabled audiotoolbox && check_apple_framework CoreAudioToolbox
 enabled avfoundation && check_apple_framework AVFoundation
 enabled coreimage    && check_apple_framework CoreImage
 enabled metal        && check_apple_framework Metal
# Tjayz
