--- configure.bak	2020-05-11 18:30:54.909379626 +0000
+++ configure	2020-05-11 18:32:22.906040345 +0000
@@ -4081,7 +4081,7 @@
                 # http://www.mingw.org
 		shared=false
      	        CFLAGS="$CFLAGS -DCST_NO_SOCKETS -DUNDER_WINDOWS -DWIN32 -shared"
-                MINGWPREF="i386-mingw32-"
+                MINGWPREF="${host_alias}-"
                 CC="${MINGWPREF}gcc"
                 RANLIB="${MINGWPREF}ranlib"
                 AR="${MINGWPREF}ar"
--- main/Makefile.bak	2020-05-11 20:18:26.039219862 +0000
+++ main/Makefile	2020-05-11 20:19:09.207549064 +0000
@@ -133,12 +133,12 @@
 
 install:
 #       The basic binaries
-	@ $(INSTALL) -m 755 $(BINDIR)/flite $(DESTDIR)$(INSTALLBINDIR)
+	@ $(INSTALL) -m 755 $(BINDIR)/flite$(EXEEXT) $(DESTDIR)$(INSTALLBINDIR)
 	@ for i in $(VOICES) ; \
 	do \
-	   $(INSTALL) $(BINDIR)/flite_$$i $(DESTDIR)$(INSTALLBINDIR); \
+	   $(INSTALL) $(BINDIR)/flite_$$i$(EXEEXT) $(DESTDIR)$(INSTALLBINDIR); \
 	done
-	$(INSTALL) -m 755 $(BINDIR)/flite_time $(DESTDIR)$(INSTALLBINDIR)
+	$(INSTALL) -m 755 $(BINDIR)/flite_time$(EXEEXT) $(DESTDIR)$(INSTALLBINDIR)
 #       The libraries: static and shared (if built)
 	cp -pd $(flite_LIBS_deps) $(DESTDIR)$(INSTALLLIBDIR)
 ifdef SHFLAGS
--- include/cst_file.h.bak	2020-05-11 18:34:47.403125914 +0000
+++ include/cst_file.h	2020-05-11 19:18:42.268404656 +0000
@@ -99,6 +99,9 @@
 int cst_fprintf(cst_file fh, const char *fmt, ...);
 int cst_sprintf(char *s, const char *fmt, ...);
 #ifdef _WIN32
+#include <windows.h>
+/* Not needed if this is mingw-w64 */
+#ifndef __MINGW64_VERSION_MAJOR
 #define snprintf c99_snprintf
 
 __inline int c99_vsnprintf(char* str, size_t size, const char* format,
@@ -121,6 +124,7 @@
        return count;
    }
 #endif
+#endif
 #define cst_snprintf snprintf
 
 #if defined(__palmos__)
--- ./src/utils/cst_tokenstream.c	2017-10-21 08:01:07.000000000 -0600
+++ ../cst_tokenstream.c	2024-01-20 23:42:18.301601000 -0700
@@ -267,7 +267,7 @@
 }
 
  #ifdef _WIN32
- __inline int ts_utf8_sequence_length(char c0)
+ int ts_utf8_sequence_length(char c0)
  #else
  int ts_utf8_sequence_length(char c0)
  #endif
--- ./src/utils/cst_val.c.bak	2024-01-20 23:54:46.864354500 -0700
+++ ./src/utils/cst_val.c	2024-01-20 23:55:00.448412300 -0700
@@ -476,7 +476,7 @@
 }
 
 #ifdef _WIN32
-__inline int utf8_sequence_length(char c0)
+int utf8_sequence_length(char c0)
 #else
 int utf8_sequence_length(char c0)
 #endif
