diff --git a/lib/gettimeofday.c b/lib/gettimeofday.c
index 36c7920af..fdc236883 100644
--- a/gl/gettimeofday.c
+++ b/gl/gettimeofday.c
@@ -35,7 +35,7 @@
 # undef LoadLibrary
 # define LoadLibrary LoadLibraryA
 
-# if !(_WIN32_WINNT >= _WIN32_WINNT_WIN8)
+# if (_WIN32_WINNT >= _WIN32_WINNT_WIN8)
 
 /* Avoid warnings from gcc -Wcast-function-type.  */
 #  define GetProcAddress \
@@ -94,7 +94,7 @@ gettimeofday (struct timeval *restrict tv, void *restrict tz)
      <http://www.windowstimestamp.com/description>.  */
   FILETIME current_time;
 
-# if !(_WIN32_WINNT >= _WIN32_WINNT_WIN8)
+# if (_WIN32_WINNT >= _WIN32_WINNT_WIN8)
   if (!initialized)
     initialize ();
 # endif
--- a/src/gl/gettimeofday.c
+++ b/src/gl/gettimeofday.c
@@ -35,7 +35,7 @@
 # undef LoadLibrary
 # define LoadLibrary LoadLibraryA
 
-# if !(_WIN32_WINNT >= _WIN32_WINNT_WIN8)
+# if (_WIN32_WINNT >= _WIN32_WINNT_WIN8)
 
 /* Avoid warnings from gcc -Wcast-function-type.  */
 #  define GetProcAddress \
@@ -94,7 +94,7 @@ gettimeofday (struct timeval *restrict tv, void *restrict tz)
      <http://www.windowstimestamp.com/description>.  */
   FILETIME current_time;
 
-# if !(_WIN32_WINNT >= _WIN32_WINNT_WIN8)
+# if (_WIN32_WINNT >= _WIN32_WINNT_WIN8)
   if (!initialized)
     initialize ();
 # endif

