diff --git a/caca/figfont.c b/caca/figfont.c
index 4c50b4e..a41966a 100644
--- a/caca/figfont.c
+++ b/caca/figfont.c
@@ -635,29 +635,3 @@ static uint32_t hsmush(uint32_t ch1, uint32_t ch2, int rule)
 
     return 0;
 }
-
-/*
- * Functions for the mingw32 runtime
- */
-
-#if defined _WIN32 && defined __GNUC__ && __GNUC__ >= 3
-#   if !HAVE_SPRINTF_S
-int sprintf_s(char *s, size_t n, const char *fmt, ...)
-{
-    va_list args;
-    int ret;
-    va_start(args, fmt);
-    ret = vsnprintf(s, n, fmt, args);
-    va_end(args);
-    return ret;
-}
-#   endif
-
-#   if !HAVE_VSNPRINTF
-int vsnprintf(char *s, size_t n, const char *fmt, va_list ap)
-{
-    return 0;
-}
-#   endif
-#endif
-
diff --git a/caca/string.c b/caca/string.c
index b2a67ce..c6dc886 100644
--- a/caca/string.c
+++ b/caca/string.c
@@ -36,16 +36,6 @@
 #include "caca.h"
 #include "caca_internals.h"
 
-#if defined _WIN32 && defined __GNUC__ && __GNUC__ >= 3
-#   if !HAVE_VSNPRINTF_S
-int vsnprintf_s(char *s, size_t n, size_t c,
-                const char *fmt, va_list ap) CACA_WEAK;
-#   endif
-#   if !HAVE_VSNPRINTF
-int vsnprintf(char *s, size_t n, const char *fmt, va_list ap) CACA_WEAK;
-#   endif
-#endif
-
 /** \brief Set cursor position.
  *
  *  Put the cursor at the given coordinates. Functions making use of the
@@ -349,13 +339,7 @@ int caca_vprintf(caca_canvas_t *cv, int x, int y, char const *format,
         buf = malloc(bufsize);
     }
 
-#if defined(HAVE_VSNPRINTF_S)
-    vsnprintf_s(buf, bufsize, _TRUNCATE, format, args);
-#elif defined(HAVE_VSNPRINTF)
-    vsnprintf(buf, bufsize, format, args);
-#else
     vsprintf(buf, format, args);
-#endif
     buf[bufsize - 1] = '\0';
 
     ret = caca_put_str(cv, x, y, buf);
@@ -609,24 +593,3 @@ int caca_set_canvas_boundaries(caca_canvas_t *cv, int x, int y, int w, int h)
 
     return 0;
 }
-
-/*
- * Functions for the mingw32 runtime
- */
-
-#if defined _WIN32 && defined __GNUC__ && __GNUC__ >= 3
-#   if !HAVE_VSNPRINTF_S
-int vsnprintf_s(char *s, size_t n, size_t c, const char *fmt, va_list ap)
-{
-    return vsnprintf(s, n, fmt, ap);
-}
-#   endif
-
-#   if !HAVE_VSNPRINTF
-int vsnprintf(char *s, size_t n, const char *fmt, va_list ap)
-{
-    return 0;
-}
-#   endif
-#endif
-
