diff -ur Makefile.bak Makefile
--- Makefile.bak	2017-03-26 18:34:52.921875000 +0200
+++ Makefile	2017-05-04 13:35:29.906250000 +0200
@@ -59,11 +59,9 @@
 $(PROGS): $(LIBRTMP)
 
 install:	$(PROGS)
-	-mkdir -p $(BINDIR) $(SBINDIR) $(MANDIR)/man1 $(MANDIR)/man8
+	-mkdir -p $(BINDIR) $(SBINDIR)
 	cp rtmpdump$(EXT) $(BINDIR)
 	cp rtmpgw$(EXT) rtmpsrv$(EXT) rtmpsuck$(EXT) $(SBINDIR)
-	cp rtmpdump.1 $(MANDIR)/man1
-	cp rtmpgw.8 $(MANDIR)/man8
 	@cd librtmp; $(MAKE) install
 
 clean:
diff -ur librtmp/Makefile.bak librtmp/Makefile
--- librtmp/Makefile.bak	2017-03-26 18:34:52.921875000 +0200
+++ librtmp/Makefile	2017-05-04 13:35:50.437500000 +0200
@@ -108,11 +108,10 @@
 install:	install_base $(SO_INST)
 
 install_base:	librtmp.a librtmp.pc
-	-mkdir -p $(INCDIR) $(LIBDIR)/pkgconfig $(MANDIR)/man3 $(SODIR)
+	-mkdir -p $(INCDIR) $(LIBDIR)/pkgconfig $(SODIR)
 	cp amf.h http.h log.h rtmp.h $(INCDIR)
 	cp librtmp.a $(LIBDIR)
 	cp librtmp.pc $(LIBDIR)/pkgconfig
-	cp librtmp.3 $(MANDIR)/man3
 
 install_so:	librtmp$(SO_EXT)
 	cp librtmp$(SO_EXT) $(SODIR)
