diff --git a/libvmaf/src/feature/adm_options.h b/libvmaf/src/feature/adm_options.h
index 72f8fa7..0b4ab75 100644
--- a/libvmaf/src/feature/adm_options.h
+++ b/libvmaf/src/feature/adm_options.h
@@ -32,5 +32,6 @@
 
 /* Whether to perform division by reciprocal-multiplication. */
 #define ADM_OPT_RECIP_DIVISION
+#undef ADM_OPT_RECIP_DIVISION
 
 #endif /* ADM_OPTIONS_H_ */
diff --git a/third_party/libsvm/Makefile b/third_party/libsvm/Makefile
index 971375a..d190e49 100644
--- a/third_party/libsvm/Makefile
+++ b/third_party/libsvm/Makefile
@@ -14,7 +14,7 @@ lib: svm.o
 	else \
 		SHARED_LIB_FLAG="-shared -Wl,-soname,libsvm.so.$(SHVER)"; \
 	fi; \
-	$(CXX) $${SHARED_LIB_FLAG} svm.o -o libsvm.so.$(SHVER)
+	$(CXX) -shared  svm.o -o libsvm.dll
 
 svm-predict: svm-predict.c svm.o
 	$(CXX) $(CFLAGS) svm-predict.c svm.o -o svm-predict -lm
diff --git a/libvmaf/src/third_party/ptools/Makefile.VMAF b/libvmaf/src/third_party/ptools/Makefile.VMAF
index 0333768..2e392f5 100644
--- a/libvmaf/src/third_party/ptools/Makefile.VMAF
+++ b/libvmaf/src/third_party/ptools/Makefile.VMAF
@@ -31,9 +31,9 @@ all: libptools
 
 libptools : $(COM_OBJS)
 ifeq ($(shell uname),Darwin)
-	$(CC) $(CCFLAGS) $(COM_OBJS) -dynamiclib -install_name $(shell pwd)/libptools.dylib -o libptools.dylib
+	$(CC) $(CCFLAGS) $(COM_OBJS) -lstdc++ -shared -o libptools.dll
 else
-	$(CC) $(CCFLAGS) $(COM_OBJS) -shared -o libptools.so
+	$(CC) $(CCFLAGS) $(COM_OBJS) -lstdc++ -shared -o libptools.dll
 endif
 
 clean :
diff --git a/libvmaf/src/third_party/ptools/opencontainers_1_8_4/include/ocsynchronizer.h b/libvmaf/src/third_party/ptools/opencontainers_1_8_4/include/ocsynchronizer.h
index 0eb577a..7f22aa2 100644
--- a/libvmaf/src/third_party/ptools/opencontainers_1_8_4/include/ocsynchronizer.h
+++ b/libvmaf/src/third_party/ptools/opencontainers_1_8_4/include/ocsynchronizer.h
@@ -11,7 +11,7 @@
 #include "ocport.h"
 #include <stdlib.h>
 #include <stdio.h>
-#if defined(_MSC_VER)
+#if defined(_MSC_VER) || defined(_WIN32) || defined(__MINGW32__)
 #define HAVE_STRUCT_TIMESPEC 0
 #include <pthread.h>
 #include <winsock2.h>
@@ -215,11 +215,11 @@
       struct timespec current_time;


-#if defined(_MSC_VER)
+#if defined(_MSC_VER) || defined(_WIN32) || defined(__MINGW32__)
         __int64 wintime; GetSystemTimeAsFileTime((FILETIME*)&wintime);
-        wintime      -=116444736000000000i64;  //1jan1601 to 1jan1970
-        current_time.tv_sec  =wintime / 10000000i64;           //seconds
-        current_time.tv_nsec =wintime % 10000000i64 *100;      //nano-seconds
+        wintime      -=116444736000000000LL;  //1jan1601 to 1jan1970
+        current_time.tv_sec  =wintime / 10000000LL;           //seconds
+        current_time.tv_nsec =wintime % 10000000LL *100;      //nano-seconds
 #elif defined(__MACH__) // OS X does not have clock_gettime, use clock_get_time
       clock_serv_t cclock;
       mach_timespec_t mts;
