--- ./modules/core/include/opencv2/core/utility.hpp.origme	2019-01-12 15:41:37.440420200 -0700
+++ ./modules/core/include/opencv2/core/utility.hpp	2019-01-12 18:40:26.436588000 -0700
@@ -60,6 +60,12 @@
 
 #if !defined(_M_CEE)
 #include <mutex>  // std::mutex, std::lock_guard
+#include <mingw.mutex.h>
+#undef OPAQUE
+#undef CONST
+#undef STRICT
+#undef MM_MIN
+#undef MM_MAX
 #endif
 
 namespace cv
