diff --git a/sources/Base/IPAddress.cpp b/sources/Base/IPAddress.cpp
index 530bd59..291a428 100644
--- a/sources/Base/IPAddress.cpp
+++ b/sources/Base/IPAddress.cpp
@@ -19,7 +19,7 @@ details (or else see http://mozilla.org/MPL/2.0/).
 #include "Base/DNS.h"
 #include "Base/Util.h"
 #if defined(_WIN32)
-#include <Iphlpapi.h>
+#include <iphlpapi.h>
 #elif !defined(__ANDROID__)
 #include <ifaddrs.h>
 #endif
diff --git a/sources/Base/Net.cpp b/sources/Base/Net.cpp
index 4884276..a6f95f0 100644
--- a/sources/Base/Net.cpp
+++ b/sources/Base/Net.cpp
@@ -17,7 +17,7 @@ details (or else see http://mozilla.org/MPL/2.0/).
 #include "Base/Net.h"
 #include "Base/SocketAddress.h"
 #if defined(_WIN32)
-#include <Iphlpapi.h>
+#include <iphlpapi.h>
 #else
 #include <net/if.h>
 #include <ifaddrs.h>
