--- src/core/windows/SDL_xinput.h	2017-01-31 22:56:57.060422793 -0700
+++ src/core/windows/SDL_xinput.h.old	2017-01-31 23:08:49.698942109 -0700
@@ -110,13 +110,13 @@
     SHORT sThumbRX;
     SHORT sThumbRY;
     DWORD dwPaddingReserved;
+} XINPUT_GAMEPAD_EX2;
-} XINPUT_GAMEPAD_EX;
 
 typedef struct
 {
     DWORD dwPacketNumber;
     XINPUT_GAMEPAD_EX Gamepad;
+} XINPUT_STATE_EX2;
-} XINPUT_STATE_EX;
 
 typedef struct
 {
