diff --git a/libavformat/asfenc.c b/libavformat/asfenc.c
index b4a3ffb..f2c1362 100644
--- a/libavformat/asfenc.c
+++ b/libavformat/asfenc.c
@@ -706,9 +706,9 @@ static void flush_packet(AVFormatContext *s)
 {
     ASFContext *asf = s->priv_data;
     int packet_hdr_size, packet_filled_size;
-
+    /* edited, not fatal anymore
     av_assert0(asf->packet_timestamp_end >= asf->packet_timestamp_start);
-
+    */
     if (asf->is_streamed)
         put_chunk(s, 0x4424, s->packet_size, 0);
