diff --git a/libavcodec/mpegvideo_enc.c b/libavcodec/mpegvideo_enc.c
index 9b50386..48ce83f 100644
--- a/libavcodec/mpegvideo_enc.c
+++ b/libavcodec/mpegvideo_enc.c
@@ -954,7 +954,14 @@ static int load_input_picture(MpegEncContext *s, const AVFrame *pic_arg)
                     av_log(s->avctx, AV_LOG_ERROR,
                            "Invalid pts (%"PRId64") <= last (%"PRId64")\n",
                            pts, last);
+                    /* edited: no fatal error anymore
                     return AVERROR(EINVAL);
+                    try to guess the pts */
+                    s->user_specified_pts =
+                    pts = s->user_specified_pts + 1;
+                    av_log(s->avctx, AV_LOG_INFO,
+                        "Warning: AVFrame.pts=? trying to guess (%"PRId64")\n",
+                        pts);
                 }
 
                 if (!s->low_delay && display_picture_number == 1)
